tests/Unit/ContentTypeSnapshotTest.php

```diff
@@ -0,0 +1,52 @@
+<?php
+
+/**
+ * This file is part of the contentful-management.php package.
+ *
+ * @copyright 2015-2017 Contentful GmbH
+ * @license   MIT
+ */
+
+namespace Contentful\Tests\Unit;
+
+use Contentful\Management\ResourceBuilder;
+
+class ContentTypeSnapshotTest extends \PHPUnit_Framework_TestCase
+{
+    public function testJsonSerialize()
+    {
+        $builder = new ResourceBuilder();
+
+        $data = [
+            'sys' => [
+                'type' => 'Snapshot',
+                'snapshotEntityType' => 'ContentType',
+            ],
+            'snapshot' => [
+                'name' => 'Versioned Content Type',
+                'displayField' => 'title',
+                'fields' => [
+                    [
+                        'name' => 'Title',
+                        'id' => 'title',
+                        'type' => 'Symbol',
+                    ],
+                    [
+                        'name' => 'Description',
+                        'id' => 'description',
+                        'type' => 'Text',
+                    ],
+                ],
+                'sys' => [
+                    'id' => 'versionedContentType',
+                    'type' => 'ContentType',
+                ],
+            ],
+        ];
+
+        $entrySnapshot = $builder->buildObjectsFromRawData($data);
```

Comment:
I moved the test, now it's an integration test, which I think it's more correct than a E2E test (it's not actually calling the API, but rather testing how different parts of the SDK interact).